### PR TITLE
PLT-3398 Adding page size option to LDAP

### PR DESCRIPTION
**Pull request by crspeller**
Related EE PR: https://github.com/mattermost/enterprise/pull/36

### CODE REVIEW

File: model/config.go [hunk 1]
```diff
@@ -516,6 +517,11 @@ func (o *Config) SetDefaults() {
 		*o.LdapSettings.QueryTimeout = 60
 	}
 
+	if o.LdapSettings.MaxPageSize == nil {
+		o.LdapSettings.MaxPageSize = new(int)
+		*o.LdapSettings.MaxPageSize = 0
```
Comment (enahum) [inline model/config.go]: Question:
if zero, it will return all results?

**Comment by coreyhulen** [comment]
+1, does this need a unit test?

File: model/config.go (hunk 1 above)
Comment (crspeller) [inline model/config.go]: Correct. This is in the help text.

**Comment by crspeller** [comment]
@coreyhulen Would love to. However since from our perspective the two functions (paged vs non-paged) are identical. I am not sure how one would write a unit test to test it.

**Comment by jwilander** [comment]
+1, though needs a rebase

**Comment by crspeller** [comment]
@jwilander Done.

**Pull request merged**